### DISCUSSION TITLE
Access metadata of current page for local quickadd capture

### DISCUSSION THREAD

**Discussion by matthewturk** [chhoumann/quickadd#228] [Q&A] [answered]
I'm trying to record things as they happen, which might be kind of quickly.  The upshot is that I want to choose from a list of items (which I can compute from the metadata in the current page) and then a choice for them.  I think that I see the right way to do this:

 1. Macro that computes available values
 2. Macro that gets other metadata (i.e., some category of what the item did)
 3. Quickadd capture to the current line or to a particular section in the open file.

I think these might need to be switched around, maybe some things moved around, but I think all of them will need some kind of access to the current page, to query its metadata (and then possibly query using, for instance, the dataview API the possible items).

Is it possible to access the current page in a macro?

**Comment by renarl** [comment, marked as answer]
It seems that this example shows how to access the current page https://github.com/chhoumann/quickadd/blob/267be761efac8ee977aee4d62a4ecd834ac219d6/docs/Examples/Attachments/zettelizer.js#L4

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that. @renarl's pointer is right, and it's still the way to do this. A macro user script gets the Obsidian `app` object, so reading the active file's metadata and driving a suggester looks like:

```js
module.exports = async (params) => {
    const file = params.app.workspace.getActiveFile();
    const frontmatter = params.app.metadataCache.getFileCache(file)?.frontmatter;

    const options = Object.keys(frontmatter ?? {});
    params.variables.picked = await params.quickAddApi.suggester(options, options);
};
```

A Capture step later in the same macro can then use `{{VALUE:picked}}`. The Zettelizer example now lives at https://quickadd.obsidian.guide/docs/Examples/Macro_Zettelizer - marking renarl's comment as the answer and closing.
